src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -232,6 +276,90 @@ def get_az_version_json():
     return versions
 
 
+def show_updates_available(new_line_before=False, new_line_after=False):
+    if should_show_updates_available():
+        import os
+        _, updates_available = get_az_version_string(use_cache=True)
+        if updates_available > 0:
+            from azure.cli.core._environment import get_config_dir
+            if new_line_before:
+                logger.warning("")
+            show_updates(updates_available)
+            if new_line_after:
+                logger.warning("")
+        # Cache the last time that should show updates available
+        dir_path = os.path.join(get_config_dir(), 'version')
+        if not os.path.exists(dir_path):
+            os.makedirs(dir_path)
+        version_time_file = os.path.join(dir_path, 'showUpdatesAvailable.txt')
+        with os.fdopen(os.open(version_time_file, os.O_RDWR | os.O_CREAT | os.O_TRUNC, 0o600),
+                       'w+') as time_file:
+            from datetime import datetime
+            time_file.write(str(datetime.now()))
```

Comment:
How about combining `showUpdatesAvailable.txt` and `azVersions.json` to one file `~/.azure/versionCheck.json` to make the file system structure simpler? Then we use the content from `lastShownTime` instead of the modification time, as the modification time may change if some one edits this file manually.

Also we can use `Session` to easily access it, like https://github.com/Azure/azure-cli/blob/7a087b14d59296832cbbabe496bef2e0a2557e28/src/azure-cli-core/azure/cli/core/_session.py#L100

Then its contents can be easily accessed with item access https://github.com/Azure/azure-cli/blob/949aaa074c569e31990e3a338a43201ddde17686/src/azure-cli-core/azure/cli/core/_profile.py#L440

Reply:
Good suggestion. Updated.